sd.go

```diff
@@ -0,0 +1,128 @@
+package gorillaz
+
+import (
+	"errors"
+	"fmt"
+	"github.com/hashicorp/consul/api"
+	"go.uber.org/zap"
+	"google.golang.org/grpc/resolver"
+	"strings"
+	"time"
+)
+
+type AgentServiceRegistrationOpt func(sd *api.AgentServiceRegistration)
+
+type ServiceDefinition struct {
+	ServiceName                    string
```

Comment:
Do you think it makes sense to replace the definition of `Addr`, `Port`, `Tags` and `Meta` here with embedding a ServiceEndpoint?

Reply:
+1